packages/platform-ios/native_modules.rb

```diff
@@ -1,7 +1,23 @@
+# This is a function which is used inside your Podfile.
+# It uses `react-native config` to grab a list of dependencies, and pulls out.all of the ones
+# which declare themselves to be iOS dependencies (via having a Podspec) and automatically
+# imports those into your current target.
+#
 def use_native_modules!(packages = nil)
   if (!packages)
+    # Resolve the CLI's main index file
     cli_bin = Pod::Executable.execute_command("node", ["-e", "console.log(require.resolve('@react-native-community/cli/build/index.js'))"], true).strip
-    output = Pod::Executable.execute_command("node", [cli_bin, "config"], true)
+    # Use the path from ^ to get the root of the node modules
+    root = cli_bin.split("node_modules/@react-native-community/cli/build/index.js").first
```

Comment:
so this works, but it's pretty brittle. Because of the hardcoded path it's now impossible to e.g. use yarn-linked cli (which is helpful for development)

I've played a bit with this, and we could take advantage of `use_native_modules!` taking parameters. I'm pretty sure in most projects the root of the project is gonna be `".."` relative to `ios/` dir – if not, users could configure this easily now. 

What do you think?:

```diff
diff --git a/packages/platform-ios/native_modules.rb b/packages/platform-ios/native_modules.rb
index 26b61d7..cfec69b 100644
--- a/packages/platform-ios/native_modules.rb
+++ b/packages/platform-ios/native_modules.rb
@@ -3,14 +3,10 @@
 # which declare themselves to be iOS dependencies (via having a Podspec) and automatically
 # imports those into your current target.
 #
-def use_native_modules!(packages = nil)
+def use_native_modules!(root = "..", packages = nil)
   if (!packages)
     # Resolve the CLI's main index file
     cli_bin = Pod::Executable.execute_command("node", ["-e", "console.log(require.resolve('@react-native-community/cli/build/index.js'))"], true).strip
-    # Use the path from ^ to get the root of the node modules
-    root = cli_bin.split("node_modules/@react-native-community/cli/build/index.js").first
-
-    throw "Auto-linking could not figure out the root folder of your package" unless root

     output = ""
     # Make sure `react-native config` is ran from your project root

```

Reply:
cc @alloy

Reply:
Alloy is out for two weeks (ish), I agree with you - I'll update this PR and add docs somewhere 👍

Reply:
Shipped, thanks!